specgen/specgen/main.go

```diff
@@ -0,0 +1,117 @@
+// Command specgen provides a utility for generating utls.ClientHelloSpecs. This command starts an
+// HTTPS server which responds to every request with a ClientHelloSpec corresponding to the TLS
+// ClientHello the server received. The spec is logged for every request as well. Make a request of
+// the server via web browser to see a ClientHelloSpec for the browser.
+//
+// It is recommended that mkcert be used to generate the keypair for this server. This ensures
+// browsers do not flag and block the page. See https://github.com/FiloSottile/mkcert.
```

Comment:
You could do this in-process, including installing it in the local trust store (which is what mkcert does anyway)

Reply:
That's an interesting idea.  However, I do not want to dig into everything that `mkcert` has to do to make that work in a cross-platform and cross-browser way (take a look at the `mkcert` code and README).  Additionally, making the user provide the certificate socializes the browser-verification issue.  When the user does run into a problem, they already know where to start, and the solution is more straightforward than making code modifications.

Reply:
Yeah, true, I like the idea of letting the user know that it's going to be a selfsigned cert (because they have to provide it).